test/src/System/Vector2.cpp

```diff
@@ -145,4 +146,47 @@ TEST_CASE("sf::Vector2 class template", "[system]")
             CHECK_FALSE(firstEqualVector != secondEqualVector);
         }
     }
+
+    SECTION("Structured bindings")
+    {
+        sf::Vector2i vector(1, 2);
+
+        SECTION("destructure by value")
+        {
+            auto [x, y] = vector;
```

Comment:
Is this something, that can be broken by our code, meaning are we testing here our code or C++'s structured binding feature?

Reply:
We are ensuring that the way `sf::Vector` is implemented allows usage of structured bindings. Turning `sf::Vector` from an aggregate into something else (e.g. by adding a `private` data member) would break this code. It's a sanity check for something that should always work, but could potentially be broken in a very unlikely scenario.